### PR TITLE
feat(telemetry): add JS package

### PR DESCRIPTION
**Pull request by tknickman**
### Description

Adds a `turbo-telemetry` library that enables tracking telemetry events from javascript packages. 

**Important**
This shares all telemetry state with turbo core's telemetry. This means users only need to disable once across all packages.

### CODE REVIEW

File: packages/turbo-telemetry/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# `@turbo/telemetry`
```
Comment (mehulkar) [inline packages/turbo-telemetry/README.md:1]: Can you include some usage instructions?

File: packages/turbo-telemetry/src/cli.ts
Comment (mehulkar) [inline packages/turbo-telemetry/src/cli.ts:1]: Is this meant to be used as a CLI or internally in other JS things (like turbo-ignore, create-turbo)? Why does it have its own CLI?

Comment (tknickman) [reply packages/turbo-telemetry/src/cli.ts:1]: This isn't a standalone CLI, it's just a helper so other CLIs can add the command without duplicating code. For example `create-turbo` (or any of our other CLIs) can just do:

```
import { Command, Option } from "commander";
import {  withTelemetryCommand } from "@turbo/telemetry";
...

const createTurboCli = new Command();

// create-turbo specific commands...

// add telemetry command
withTelemetryCommand(createTurboCli);
```

File: packages/turbo-telemetry/README.md (hunk 1 above)
Comment (tknickman) [reply packages/turbo-telemetry/README.md:1]: Yes, will do!

File: packages/turbo-telemetry/src/cli.ts
Comment (tknickman) [reply packages/turbo-telemetry/src/cli.ts:1]: Add the command to other CLIs that use this allows telemetry to be disabled without needing `turbo` core. Just incase we have clients that consume a package, without `turbo` itself.

**Review by chris-olszewski — APPROVED** [review]
LGTM. I do have some slight concern about keeping the event keys in sync. There's probably something we could do with `napi`,  but no need to build infra before we need it.

File: packages/turbo-telemetry/src/utils.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+export function buildUserAgent(packageInfo: PackageInfo): string {
+  const nodeVersion = process.version;
+  const operatingSystem = os.type();
+  const architecture = os.arch();
+
+  return `${packageInfo.name} ${packageInfo.version} ${nodeVersion} ${operatingSystem} ${architecture}`;
+}
+
+export async function defaultConfigPath() {
+  const dir = await configDir();
```
[the remark this answers is not in the document]
Comment (tknickman) [reply packages/turbo-telemetry/src/utils.ts:15]: We shouldn't _need_ that here, but I'll add it just to match

**Comment by tknickman** [comment]
@chris-olszewski the intention isn't to mirror events here one to one, but for the events tracked in the JS packages to be unique and specific to the JS packages. These also use a different event type on the backend that always tracks the package name and version as context. 


I did think about napi for some of the shared code (like config) but the lift was low and didn't want to deal with the multi os / arch support here.

**Pull request merged**